youtube_dl/extractor/streamcz.py

```diff
@@ -20,13 +21,21 @@ def _get_api_key(api_path):
     return hashlib.md5(a.encode('ascii')).hexdigest()
 
 
-class StreamCZIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?stream\.cz/.+/(?P<id>[0-9]+)'
+class StreamCZBaseIE(InfoExtractor):
     _API_URL = 'http://www.stream.cz/API'
 
+    def _get_api_request(self, api_path, api_key):
+        req = sanitized_Request(self._API_URL + api_path)
+        req.add_header('Api-Password', _get_api_key(api_path))
+        return req
+
+
+class StreamCZIE(StreamCZBaseIE):
+    _VALID_URL = r'(?:https?://(?:www\.)?stream\.cz/(?!porady|pohadky).+/|streamcz:)(?P<id>[0-9]+)'
```

Comment:
That's incorrect. https://www.stream.cz/pohadky/lego-staveni/10015386-dalsi-staveni-z-lego-city-starter-set is a video but processed as playlist.